x/evm/abci.go

```diff
@@ -65,25 +65,40 @@ func handleTokenSent(ctx sdk.Context, event types.Event, bk types.BaseKeeper, n
 }
 
 func handleContractCall(ctx sdk.Context, event types.Event, bk types.BaseKeeper, n types.Nexus, multisig types.MultisigKeeper) error {
-	e := event.GetEvent().(*types.Event_ContractCall).ContractCall
+	e := event.GetContractCall()
 	if e == nil {
 		panic(fmt.Errorf("event is nil"))
 	}
 
 	destinationChain := funcs.MustOk(n.GetChain(ctx, e.DestinationChain))
-	destinationCk := funcs.Must(bk.ForChain(ctx, destinationChain.Name))
+	switch destinationChain.Module {
+	case types.ModuleName:
+		handleContractCallToEVM(ctx, bk, multisig, n, destinationChain.Name, event)
+		return nil
+	default:
+		// set as general message in nexus, so the dest module can handle the message
+		return setGeneralMessageToNexus(ctx, n, event)
```

Comment:
Longer term this should change to calling two nexus methods.
- ConfirmMessage to set a message in nexus
- RouteMessage to route it to the destination module, which will call the evm handler to add it to the command queue